### DISCUSSION TITLE
why it occurs??

### DISCUSSION THREAD

**Discussion by Coderyang666666** [monarc-project/MonarcAppFO#527] [Q&A] [answered]
[Fri Mar 15 18:27:01.621343 2024] [php:notice] [pid 1052] [client ::1:31335] PHP Fatal error:  Uncaught Error: Call to a member function has() on null in /home/yanghang/vendor/laminas/laminas-modulemanager/src/Listener/ServiceListener.php:198\nStack trace:\n#0 /home/yanghang/vendor/laminas/laminas-eventmanager/src/EventManager.php(319): Laminas\\ModuleManager\\Listener\\ServiceListener->onLoadModulesPost()\n#1 /home/yanghang/vendor/laminas/laminas-eventmanager/src/EventManager.php(169): Laminas\\EventManager\\EventManager->triggerListeners()\n#2 /home/yanghang/vendor/laminas/laminas-modulemanager/src/ModuleManager.php(109): Laminas\\EventManager\\EventManager->triggerEvent()\n#3 /home/yanghang/vendor/laminas/laminas-mvc/src/Application.php(261): Laminas\\ModuleManager\\ModuleManager->loadModules()\n#4 /home/yanghang/public/index.php(55): Laminas\\Mvc\\Application::init()\n#5 {main}\n  thrown in /home/yanghang/vendor/laminas/laminas-modulemanager/src/Listener/ServiceListener.php on line 198

**Comment by Coderyang666666** [comment]
if (! $this->defaultServiceManager->has($sm['service_manager']))

**Comment by ruslanbaidan** [comment]
Please, provide more context on: where it happens (env) and how and to reproduce.
In general this is the application dependencies issue, not related to the app itself and can be solved by a proper installation.

**Comment by Coderyang666666** [comment]
url input localhost, error.log this fatal error appears, installFO.sh has been executed. ServiceListener.php is at 
  \home\yanghang\vendor\laminas\laminas-modulemanager\src\Listener\ServiceListener.php. Class ServiceListener's Constructor function is public function __construct(  
        /** 
         * Default service manager used to fulfill other SMs that need to be lazy loaded
         */
        ServiceManager $defaultServiceManager,
        $configuration = null
    ) 
    {
        if ($configuration !== null) 
        {
            $this->setDefaultServiceConfig($configuration);
        }
    }  
$this->setDefaultServiceConfig($configuration);It doesn't seem to be being executed
The control flow is executed to 
if (! $sm['service_manager'] instanceof ServiceManager) 
            {
                xdebug_break(); 
                
                if (! $this->defaultServiceManager->has($sm['service_manager']))
                
                {
                    // No plugin manager registered by that name; nothing to configure.
                    continue;
                }
$this->defaultServiceManager- = null?
the Stack trace has ben already explicitly above.
   I would have liked to debug, but I wouldn't debug with xdebug on apache2
The help is greatly appreciated

**Reply by Coderyang666666** [reply]
Like you said, I don't know which function calls the class ServiceListener

**Reply by Coderyang666666** [reply]
php's version is 8.1 at wsl
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/95f684cd-d9f2-4866-a26f-01b99c5e1b29)

**Reply by ruslanbaidan** [reply, marked as answer]
The result `php -v` command shows already a lot of installation problems.
For example without pdo_msql the app can't work, but we are not advising on the system installations.
ServiceListener is the external Laminas (PHP framework used in Monarc) dependency and there is a problem of the project installation, but nothing to do the the application itself.

**Reply by Coderyang666666** [reply]
Thanks, I'll dig into that according this question

**Reply by Coderyang666666** [reply]
Why is it displayed like this， after http://localhost 
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/fe1c102e-52cf-4581-add5-7b13db54d064)

**Reply by Coderyang666666** [reply]
Under the http://localhost page
![Uploading image.png…]()

**Reply by ruslanbaidan** [reply]
The way of Monarc installation is to contribute to the development. Is this the purpose ?
Otherwise please use the VM https://www.monarc.lu/download/#virtual-machine

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/296f73f3-db38-4fdd-ba82-cc180f88697d)
What is the initial email address and password to Log in to the front-end and back-end？?

**Reply by ruslanbaidan** [reply]
All of the information available in the each VM release readme file.

**Reply by Coderyang666666** [reply]
But I can't log in successfully with this username and password.Thank you for your continued help to me!!!I also hope to be able to contribute to this project with your help.
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/b1b22772-1fff-41ab-acc7-b6d6baaa43c5)

![Uploading image.png…]()

**Reply by Coderyang666666** [reply]
![Uploading image.png…]()

**Reply by Coderyang666666** [reply]
I tried all kinds to find the username and password to read my file, but I still can't log in, I hope to get help, thanks. on the basis of " The username is *admin@admin.localhost* and the password is *admin*." in readme file 
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/63173c64-5fb5-4685-9a53-db3fb37eaa77)

**Reply by ruslanbaidan** [reply]
You can try in another browser or clean cache.
If this is VM then it should be the user. If manual installation, then can be an installation or configuration issue, but if properly doebn the user login and pass are the same as on VM.

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/0f376a53-6f13-4e46-800b-27baced1fad6)
When this sentence is located, the terminal shows that the execution is successful, but it cannot log in
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/5fbc039c-bcc1-47c3-b421-ccf5056f495c)

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/69f70b7f-454a-48ab-8a6f-bbcfc1ad7cff)
And when I enter my password in this interface, I can't receive the email either

**Reply by Coderyang666666** [reply]
I installed it successfully under WSL，ubuntu 20.04

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/02eadc19-5a01-4025-81c7-964178029ad4)
Why am I blank here and can't select a model?

**Reply by ruslanbaidan** [reply]
Check your browser dev. console, database, apache error logs etc.
There could be many reasons.
The solution, as it was said, **usage the VM**.

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/3fee0b74-f130-44fb-bf1a-fedb9bdae40d)
The link corresponding to this parameter cannot be opened.How to fix it ?
![Uploading image.png…]()

**Reply by Coderyang666666** [reply]
![Uploading image.png…]()

**Reply by Coderyang666666** [reply]
It's strange that I've come this with this mistake.
at  \var\log\apache2\error.log:
Mon Apr  1 16:55:37 2024 (5087): Error Internal error: wrong size calculation: /home/yanghang/MonarcAppFO/vendor/monarc/frontoffice/src/Controller/ApiAnrScalesCommentsControllerFactory.php start=0x000000007aaf61c0, end=0x000000007aaf6890, real=0x000000007aaf68f0

[Mon Apr 01 16:55:38.253462 2024] [core:notice] [pid 2760] AH00051: child pid 2765 exit signal Segmentation fault (11), possible coredump in /etc/apache2

![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/5b2b0749-889f-4897-8e0a-83d1b76917f5)

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/7eab18d6-a92d-40a4-9208-969466d87406)
I ask for one more question，Where exactly are files like /api/stats/validate-stats-availability and /api/client-anr?I didn't find it locally at all.

**Reply by Coderyang666666** [reply]
All your rights have been revoked. Please contact your organization's manager. Why is this prompted when I log in?
use
user: admin@admin.localhost / password: admin

**Reply by Coderyang666666** [reply]
![image](https://github.com/monarc-project/MonarcAppFO/assets/90095020/c9d9e849-83e0-4772-af99-7897a0baaa89)
